pmd-java/src/test/resources/net/sourceforge/pmd/ast/ParserCornerCases18.java

```diff
@@ -165,6 +165,27 @@ public void lambdaWithPropertyAssignment() {
         }
     }
 
+    public List<@AnnotatedUsage ?> testWildCardWithAnnotation() {
+    	return null;
+    }
+    
+	public Object @Nullable [] testAnnotationsToArrayElements() {
+		return null;
+	}
+	
+	private byte @Nullable [] getBytes(){
+		return null;
+	}
+	
+	public static <T extends @NonNull Enum<T>> getEnum() {
+		return null;
+	}
+	
+	public void createNonNullArray() {
+		return new MetaColumn @NonNull [value];
```

Comment:
your grammar update supports it, but I'd love to see a multi-dimensional array here, each dimension with it's own annotation